### PR TITLE
Change the API for Private Cluster as per new agreed model

### LINKED ISSUE #2024: Private Cluster - Public API
Proposing to add a **NEW** field named `APIServerProperties` into the existing `MasterPoolProfile`.

Having a new dedicated structure for all current and future API customizations will help to be more flexible and not overload the existing properties struct.

```
// MasterPoolProfile contains configuration for OpenShift master VMs.
type MasterPoolProfile struct {
	Count      *int64  `json:"count,omitempty"`
	VMSize     *VMSize `json:"vmSize,omitempty"`
	SubnetCIDR *string `json:"subnetCidr,omitempty"`
	// CertsProfile (in, optional): configures OpenShift Master API certificates 
	CertsProfile *CertsProfile `json:"certsProfile,omitempty"`
	APIProperties *APIProperties `json:"APIProperties,omitempty"`	
}

// CertsProfile contains configuration for OpenShift certificates. 
type CertsProfile struct { 
	// KeyVaultSecretID (in, optional): the customer secret ID certificate used 
	KeyVaultSecretID *string `json:"keyVaultSecretID,omitempty"` 
	// CertificateExpirationDate (out): the expiration date of the current certificate used 
	CertificateExpirationDate *string `json:"CertificateExpirationDate,omitempty"` 
}

type APIProperties struct {
	// PrivateAPIServer (in, optional): Specifies if API server is public or private
	PrivateAPIServer *bool `json:"privateApiServer,omitempty"`

	// This can provide more customization for future implementation such as :
	// VanityDomain (in, optional): define the custom root domain to use for the api server
	VanityDomain *string `json:"vanityDomain,omitempty"`
}
```

### PR DESCRIPTION
**Pull request by asalkeld**
```release-note
Change the API for Private Cluster as per new agreed model
```
closes #2024 
/cc @ehashman 
@julienstroheker Hopefully this is what you are after.

### CODE REVIEW

File: pkg/api/2019-10-27-preview/types.go [hunk 1]
```diff
@@ -128,11 +128,19 @@ type RouterProfile struct {
...
+	APIProperties *APIProperties `json:"apiProperties,omitempty"`
+}
+
+// APIProperties defines further properties on the API
+type APIProperties struct {
+	// PrivateAPIServer (in, optional): Specifies if API server is public or private
 	PrivateAPIServer *bool `json:"privateApiServer,omitempty"`
+
+	// VanityDomain (in, optional): define the custom root domain to use for the api server
+	VanityDomain *string `json:"vanityDomain,omitempty"`
```
Comment (jim-minter) [inline pkg/api/2019-10-27-preview/types.go]: how does this field differ from properties.fqdn ?

Comment (asalkeld) [reply pkg/api/2019-10-27-preview/types.go]: Not sure, :confused: I am inclined to remove it until we use it. (or i could just comment it out)

Comment (julienstroheker) [reply pkg/api/2019-10-27-preview/types.go]: This can be removed

File: pkg/api/2019-10-27-preview/convertertointernal.go [hunk 2]
```diff
@@ -90,8 +90,10 @@ func mergeProperties(oc *OpenShiftManagedCluster, cs *api.OpenShiftManagedCluste
 	if oc.Properties.FQDN != nil {
 		cs.Properties.FQDN = *oc.Properties.FQDN
 	}
-	if oc.Properties.MasterPoolProfile != nil && oc.Properties.MasterPoolProfile.PrivateAPIServer != nil {
-		cs.Properties.PrivateAPIServer = *oc.Properties.MasterPoolProfile.PrivateAPIServer
+	if oc.Properties.MasterPoolProfile != nil && oc.Properties.MasterPoolProfile.APIProperties != nil {
+		if oc.Properties.MasterPoolProfile.APIProperties.PrivateAPIServer != nil {
+			cs.Properties.PrivateAPIServer = *oc.Properties.MasterPoolProfile.APIProperties.PrivateAPIServer
```
Comment (julienstroheker) [inline pkg/api/2019-10-27-preview/convertertointernal.go]: This internal field stay under the properties structure ? Why not also moving it to the `MasterPoolProfile` ?

Comment (asalkeld) [reply pkg/api/2019-10-27-preview/convertertointernal.go]: this is the internal API

Comment (asalkeld) [reply pkg/api/2019-10-27-preview/convertertointernal.go]: (the internal API doesn't have a MasterPoolProfile)

**Comment by ehashman** [comment]
flake #1715 on e2e-create-20191027-private
flake #552 on upgrade-v7.1
flake #2032 on upgrade-v10.0

/retest

**Comment by ehashman** [comment]
/hold

Waiting to fix CI

**Pull request merged**